snowpark_style_guide.md

```diff
@@ -0,0 +1,377 @@
+# Snowpark Style Guide
+
+The Snowpark library provides intuitive APIs for querying and processing data in a data pipeline.
+Using this library, you can build applications that process data in Snowflake without having to move data to the system where your application code runs.
+
+This guide to Snowpark code style demonstrates common situations we've encountered and the associated best practices based on our experience coaching and working with our engineering teams.
+
+Beyond Snowpark specifics, the general practices of clean code are important in Snowpark -- the
+[PEP 8][pep-8] is a strong starting point for learning more about these practices.
+
+# Table of Contents
+
+1. [DataFrame Operations](#dataframe-operations)
+   1. [Column Selection](#column-selection)
+   2. [Aliasing Column Name](#aliasing-column-name)
+   3. [Performant SQL Generation](#performant-sql-generation)
+2. [Expressions](#expressions)
+   1. [Multiple-Lines](#multiple-lines)
+3. [Explicitness](#explicitness)
+   1. [Calling Methods and Functions with Parameter Names](#calling-methods-and-functions-with-parameter-names)
+4. [User Defined Functions (UDFs)](#user-defined-functions-udfs)
+   1. [Registration](#registration)
+   2. [Type Hints](#type-hints)
+5. [User Defined Table Functions (UDTFs)](#user-defined-table-functions-udtfs)
+6. [Stored Procedures](#stored-procedures)
+6. [Miscellaneous](#miscellaneous)
+
+# DataFrame Operations
+
+## Column Selection
+
+In Snowpark, to provide the best development experience for developers coming from various background,
+multiple approaches of column selection are provided for handy
+usage as demonstrated in the following snippets:
+```python
+# select by attribute
+df = df.select(df.col_a, df.col_b)
+
+# select by string
+df = df.select("col_a", "col_b")
+
+# select by using snowflake.snowpark.functions.col function
+df = df.select(col("col_a"), col("col_b"))
+
+# select by using snowflake.snowpark.functions.lit function
+df = df.select(lit("col_a"), lit("col_b"))
```

Comment:
hmm `lit` is used for select a constant python value like `lit(2)`, `lit("x")`, `lit([1,2])`

Reply:
updated